src/main/kotlin/pl/edu/agh/gem/internal/job/stage/FindSettlementsStage.kt

```diff
@@ -5,13 +5,13 @@ import pl.edu.agh.gem.internal.job.ProcessingStage
 import pl.edu.agh.gem.internal.job.ReconciliationJobState.SAVING
 import pl.edu.agh.gem.internal.job.StageResult
 import pl.edu.agh.gem.internal.model.reconciliation.ReconciliationJob
-import pl.edu.agh.gem.internal.solver.settlements.GreedySettlementsSolver
+import pl.edu.agh.gem.internal.solver.settlements.CombinedSettlementSolver
 
 @Component
 class FindSettlementsStage : ProcessingStage() {
     override fun process(reconciliationJob: ReconciliationJob): StageResult {
         logger.info { "Finding settlements for job: $reconciliationJob" }
-        val settlements = GreedySettlementsSolver.solve(reconciliationJob.balances)
+        val settlements = CombinedSettlementSolver.solve(reconciliationJob.balances)
```

Comment:
i am dont sure if we want to take that aproach in case of any retry we will run code of combinedSettlementSolver each time, we have Stages so we can move that code to one of the stages and then decide what stage of Solver choose